### ISSUE TITLE
Training freezes after adding new training examples.

### ISSUE THREAD

**Issue by Dylan0126** [RasaHQ/rasa_nlu#1490]
**Rasa NLU version**: 0.11.3
Scikit-Learn : 0.20.0

**Operating system** (windows, osx, ...): Amazon Linux (EC2)   m5d.4xlarge
16vCPU    64G Memory

**Content of model configuration file**:
{
  "pipeline": "spacy_sklearn",
  "path" : "./models",
  "data" : "./data/train.json",
  "spacy_model_name" : "en_core_web_lg"
}
```json

```

**Issue**: Training freezes after adding a few new training examples.

Training :
	- intent examples: 70155 (126 distinct intents)
	- entity examples: 42188 (12 distinct entities)
INFO:rasa_nlu.model:Starting to train component nlp_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component tokenizer_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component intent_featurizer_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component intent_entity_featurizer_regex
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component ner_crf
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component ner_synonyms
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component intent_classifier_sklearn
Fitting 3 folds for each of 6 candidates, totalling 18 fits
[Parallel(n_jobs=16)]: Using backend LokyBackend with 16 concurrent workers.
[Parallel(n_jobs=16)]: Done   6 out of  18 | elapsed: 38.3min remaining: 76.6min

Everything works fine until the last line above, then CPU utilization drops gradually until it reaches a low level and then training process freezes and will not finish.

<img width="732" alt="screen shot 2018-10-29 at 3 55 33 pm" src="https://user-images.githubusercontent.com/30248273/47685292-2f988400-db93-11e8-8ca6-efb3a88a79fc.png">

At time 21:17 is about when the last line showed up "Done 6 out of 18", then CPU utilization started to drop.

The training used to take about 2 hours to finish, now when it freezes the training will never end. Adding only 6 new rows of training examples, there are nothing special about the content. 

I thought it could be caused by using sklearn.svm.SVC in the Gridsearch process
So I changed the code in sklearn_intent_classifier to use LinearSVC.
Now the training can be finished in 7 minutes, but I got warnings:

 ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)

What could be a good number of iterations to use if I want to roughly replicate the performance of my old model?

**Comment by akelad** [comment]
Thanks for raising this issue, @EPedrotti  will get back to you about it soon.

**Comment by Dylan0126** [comment]
Hi @EPedrotti,
Then do you think there is a solution to solve the never ending training problem without change SVC to LinearSVC? Since we have been using the SVC one for a long time.

**Comment by EPedrotti** [comment]
@Dylan0126 maybe this can be of some help for the never ending training https://datascience.stackexchange.com/questions/989/svm-using-scikit-learn-runs-endlessly-and-never-completes-execution. If you are using LinearSVC remember to normalize you data (see link).
Also, since you have a large dataset you could also try to use the tensorflow_embedding pipeline and see how it performs.

**Comment by EPedrotti** [comment]
@Dylan0126 if you have more usage question about Rasa you could also have a look at our forum [https://forum.rasa.com/](https://forum.rasa.com/)
